<!DOCTYPE html>
<html lang="en-us"
      xmlns:og="http://ogp.me/ns#"
      xmlns:fb="https://www.facebook.com/2008/fbml">
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="author" content="Mike Hallett">
    <meta name="description" content="Explore and model biological data">

    <meta name="twitter:card" value="summary_large_image">
    <meta name="twitter:site" content="@hallettmichael">
    <meta name="twitter:title" content="Assignment 2 | BIOL 480/510/630: Bioinformatics">
    <meta name="twitter:creator" content="@hallettmichael">
    <meta name="twitter:image" content="/images/sys-small1.png">
    <meta name="twitter:description" content="Explore and model biological data" />

    <meta property="og:site_name" content="BIOL 480/510/630: Bioinformatics"/>
    <meta property="og:title" content="Assignment 2 | BIOL 480/510/630: Bioinformatics">
    <meta property="og:type" content="article" />
    <meta property="article:author" content="" />
    <meta property="og:url" content="/" />
    <meta property="og:image" content="/images/sys-small1.png">
    <meta property="og:description" content="Explore and model biological data" />

    <meta name="generator" content="Hugo 0.74.1" />    

    <title>Assignment 2 | BIOL 480/510/630: Bioinformatics</title>
    <link rel="canonical" href="/assignment/02-assignment/">
    <link href="/theme/css/tufte.css" rel="stylesheet">
    <link href="/theme/css/latex.css" rel="stylesheet">
    <link href="/theme/css/custom.css" rel="stylesheet">

    <!--[if lt IE 9]>
    <script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
    <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->

    <style type="text/css">
        .wrap a, .wrap a:visited {
            color: #1663BA;
        }

        .wrap a:hover {
            color: #398AE6;
        }

        .wrap p > a {
            color: #1663BA;
        }

        article .sidenote-number:after, article .sidenote:before {
            color: #1663BA;
        }

        article .sidenote:before {
            color: #1663BA;
        }

        @media (max-width: 760px) {
            label.margin-toggle:not(.sidenote-number) {
                color: #1663BA;
            }
        }

        .contrast a, .contrast a:visited {
            color: #1663BA; 
        }

        header > nav a {
            color: #1663BA;
        }

        .credits a {
            color: #1663BA;
        }

        .wrap .tag {
            background: #1663BA;
        }
    </style>

    <link rel="shortcut icon" href="/favicon.ico" />
    <link rel="apple-touch-icon-precomposed" sizes="57x57" href="/images/apple-touch-icon-57x57.png" />
    <link rel="apple-touch-icon-precomposed" sizes="114x114" href="/images/apple-touch-icon-114x114.png" />
    <link rel="apple-touch-icon-precomposed" sizes="72x72" href="/images/apple-touch-icon-72x72.png" />
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/images/apple-touch-icon-144x144.png" />
    <link rel="apple-touch-icon-precomposed" sizes="120x120" href="/images/apple-touch-icon-120x120.png" />
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="/images/apple-touch-icon-152x152.png" />
    <link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32" />
    <link rel="icon" type="image/png" href="/images/favicon-16x16.png" sizes="16x16" />
    <meta name="application-name" content="BIOL 480/510/630: Bioinformatics"/>
    <meta name="msapplication-TileColor" content="#FFFFFF" /> 
    <meta name="msapplication-TileImage" content="/images/mstile-144x144.png" />

    
    <script src="//use.typekit.net/vpe6zfl.js"></script>
    <script>try{Typekit.load({ async: true });}catch(e){}</script>

    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">


 

</head>

<body>
<div class="wrap">
    <header>
        <nav class="group logo-wrap">
            <a href="/" class="logo"><img src="/images/sys-small1.png" alt="Data Science in Biology"></a>
            <h1 class="websitetitle"><a href="/">BIOL 480/510/630: Bioinformatics</a></h1>
        </nav>
        <nav class="bottom-group">
            
            <a href="/overview/">Overview</a>
            
            <a href="/syllabus/">Syllabus</a>
            
            <a href="/resources/">Resources</a>
            
            <a href="/assignments/">Assignments</a>
            
            <a href="/projects/">Projects</a>
            
        </nav>
    </header>
 

    <article class="group">
    <h1>Assignment 2</h1>

   
    
    
<script src="/rmarkdown-libs/header-attrs/header-attrs.js"></script>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>


<p>Please note that due dates can be found in the <a href="/Syllabus">Syllabus</a>; submission instructions can be found on the <a href="/Assignments">Assignment Instructions</a> page. In this assignment, you can submit a Google Doc (or other text editor, pictures, etc.) but also your <span class="math inline">\({\tt R}\)</span> code via Google Drive.
Aki will go over the submission process in the lab.</p>
<p>You might consider (but it is not mandatory) using R Markdown to write your answers.</p>
<p><span class="math inline">\({\bf 50}\)</span> total marks.</p>
<p><strong>Question 1 [1 points]</strong>
Re-express the following small program using pipes.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r watch-out"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a>x&lt;-<span class="dv">3</span>; mu&lt;-<span class="dv">5</span>; sigma &lt;-<span class="dv">7</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a>nrm &lt;-<span class="st"> </span>( <span class="dv">1</span> <span class="op">/</span><span class="st"> </span>(sigma <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>( <span class="dv">2</span> <span class="op">*</span><span class="st"> </span>pi ))) <span class="op">*</span><span class="st"> </span><span class="kw">exp</span>( <span class="op">-</span>(<span class="dv">1</span><span class="op">/</span><span class="dv">2</span>) <span class="op">*</span><span class="st"> </span>( (x <span class="op">-</span><span class="st"> </span>mu)<span class="op">/</span><span class="st"> </span>sigma )<span class="op">^</span><span class="dv">2</span>)</span></code></pre></div>
<p>Hint:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r watch-out"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">c</span>( <span class="dt">first =</span> <span class="st">&quot;a&quot;</span>, <span class="dt">second =</span> <span class="st">&quot;b&quot;</span>, <span class="dt">third =</span> <span class="st">&quot;c&quot;</span>)</span></code></pre></div>
<p><em>Bonus [1/2 point]</em> What is this equation?</p>
<p><strong>Question 2 [3 points]</strong>
Re-express the following small program using pipes.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r watch-out"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a>site &lt;-<span class="st"> </span><span class="dv">15</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a><span class="kw">gsub</span>(<span class="st">&quot;;&quot;</span>, <span class="st">&quot;/&quot;</span>, <span class="kw">paste</span>(<span class="kw">paste</span>( <span class="kw">paste</span>( <span class="st">&quot;~hallett&quot;</span>, <span class="kw">paste</span>(<span class="st">&quot;raw&quot;</span>, <span class="st">&quot;tara&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>), <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>), </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true"></a>      <span class="kw">paste</span>(<span class="st">&quot;site&quot;</span>, site, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;;&quot;</span>), </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true"></a>      <span class="st">&quot;dna_seq.fa&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;;&quot;</span>))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true"></a><span class="co">## [1] &quot;~hallett/raw/tara/site_15/dna_seq.fa&quot;</span></span></code></pre></div>
<p><strong>Question 3 [6 points]</strong></p>
<p>3a [1 point] How many black or african women are included in the <span class="math inline">\({\tt small\_brca}\)</span> dataset?</p>
<p>3b [1 point] How many black or african males?</p>
<p>3c [3 points] Create a histogram of all black or african individuals grouped by age. Each bar in the histogram should have two colours corresponding to male and females. (For example, red might be used for the fraction of women in some age group while blue is used for men in that same age group). Choose the number of bins (or binsize) in a way that makes your plot informative.</p>
<p>3d [1 point] What was the rationale for the number of bins you used?</p>
<p>Show code for all parts <span class="math inline">\(3a\)</span> to <span class="math inline">\(3d\)</span>.</p>
<p><strong>Question 4 [7 points]</strong></p>
<p>4a [2 point] How many participants are there? (Careful: not samples but participants.) Show code.</p>
<p>4b [3 points] List all of the <span class="math inline">\({\tt participant}\)</span> that have more than one sample. Show code.</p>
<p>4c [2 points] For the participants identfied in 3b, create a scatter plot with the tumor on the <span class="math inline">\(x\)</span> axis and the normal on the <span class="math inline">\(y\)</span> axis (as specified by the <span class="math inline">\({\tt tumor}\)</span> variable in the tibble) with points representing the expression of gene <span class="math inline">\({\tt ESR1}\)</span>.</p>
<p>Show your code for each step.</p>
<p><strong>Question 5 [7 points]</strong></p>
<p>Create the following plot using the <span class="math inline">\({\tt small\_brca}\)</span> tibble.</p>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_point).</code></pre>
<p><img src="/assignment/02-assignment_files/figure-html/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Question 6 [4 points]</strong></p>
<p>In the lecture I said that the violin plot has killed the boxplot. <strong>Maya</strong>, your classmate, contradicted me and suggested boxplots are, actually, cool.
Well I thought then that the whole class should get another question to explore <strong>Maya’s</strong> suggestion. Let’s all thank <strong>Maya</strong>!!</p>
<p>6a [3 points] In one figure (hint: <span class="math inline">\({\tt ggarrange}\)</span>) using the log-transformed expression of <span class="math inline">\({\tt ESR1}\)</span> create (i) a boxplot, (ii) a violin plot, (iii) a boxplot with beeswarm.</p>
<p>6b [1 point] In your opinion - whatever you think now - which is the nicest?</p>
<p><strong>Question 7 [5 points]</strong></p>
<p>The variable <span class="math inline">\({\tt race}\)</span> in <span class="math inline">\({\tt small\_brca}\)</span> has several different possible values including “WHITE”, “ASIAN”, “BLACK OR AFRICAN AMERICAN”, “AMERICAN INDIAN OR ALASKA NATIVE”, “[Not Available]”, “[Not Evaluated]”, or NA.</p>
<p>It is better to simply use NA instead of“[Not Available]”, “[Not Evaluated]”, or NA.
It will simplify your code in downstream analyses.</p>
<p>Write R code to create a new <span class="math inline">\({\tt race\_modified}\)</span> variable where all samples with “[Not Available]”, “[Not Evaluated]” are changed to NA (Note: you shouldn’t use “NA”, but NA. They are not the same)</p>
<p><strong>Question 8 [5 points]</strong></p>
<p>Samples may have differences in the distributions of expression across the genes.
This might be caused by technical noise because often not all samples are processed at the
same time or the samples might be prepared slightly differently.
This difference in distributions can sometimes get in the way of our analysis, especially
when we are looking at many genes at the same time.
A common approach that sometimes helps is to <em>median center</em> the expression of each gene.
For any gene, the median centering transforms transcript counts <span class="math inline">\(G = (g_1, g_2, \ldots, g_n)\)</span>
to <span class="math inline">\(G&#39;= (g&#39;_1, g&#39;_2, \ldots, g&#39;_n)\)</span> where <span class="math inline">\(g&#39;_i = g_i - median(G)\)</span>.
Here <span class="math inline">\(n\)</span> is the number of genes (in our case <span class="math inline">\(n=50\)</span>).</p>
<p>Show code to create a new tibble where all expression measurements are median centered.</p>
<p><strong>Question 9 [7 points]</strong></p>
<p>Different genes have different distributions of expression.
We saw evidence of this in Lecture 06 (slide 25) when we created a violin plot across all <span class="math inline">\(50\)</span> of our genes.
This difference in distributions can also sometimes get in the way of our analysis, especially
when we are looking at many genes at the same time.
Or sometimes we want to only those genes that have large or highly <em>variable</em> expression.</p>
<p><strong>9a [1 point]:</strong> Using online resources, define what the <em>variance</em> of a distribution is. Find a function in R to compute the variance.</p>
<p><strong>9b [4 points]:</strong> Compute the variance for each of the <span class="math inline">\(50\)</span> genes in our dataset, create
a tibble with just the <span class="math inline">\({\tt participant}\)</span> and <span class="math inline">\({\tt race}\)</span> variables along with the <span class="math inline">\(50\)</span> genes. Order the columns (variables) of the tibble from most to least variable.</p>
<p><strong>9c [2 points]:</strong> Create a violin plot of the <span class="math inline">\(50\)</span> genes ordered from most to least variable.</p>
<p><strong>Question 10 [5 points]</strong>
Pick one of the <span class="math inline">\(50\)</span> genes from <span class="math inline">\({\tt small\_brca}\)</span> uniformly randomly (see the <span class="math inline">\({\tt runif}\)</span> function). Using the NCBI, provide the following information or note that it is not availalbe:</p>
<p>10a Full name of the gene and the official name according to the HGNC.<br>
10b First time it was reported in the literature.<br>
10c Where it is located in the genome.<br>
10d Its <span class="math inline">\({\tt gi}\)</span> acccession code or codes (if it has been modified).<br>
10e The number of exons.<br>
10f The number of alternative transcripts that have been record. <br>
10g Its full amino acid sequence. Provide its protein ID.<br>
10h Its protein structure, if known.<br></p>
<p>Good luck!</p>


    </article>

    <footer>
        <hr class="slender">
        <div class="course-info">
        <span class="course-name">
            BIOL 480/510/630: Bioinformatics (Fall 2020)
        </span><br>
        <span>
            <em><a href="https://www.concordia.ca">Concordia University</a></em> | 
            <em><a href="https://www.concordia.ca/artsci/biology.html">Department of Biology | CSFG | CASB</a></em>
        </span>
        </div>

        <ul class="footer-links">
            <li>
            <a href="https://mikehallett.science"><i class="fa fa-user" aria-hidden="true"></i> Mike Hallett</a>
            </li>

            <li>
            <a href="mailto:hallett.mike.t@gmail.com"><i class="fa fa-envelope" aria-hidden="true"></i> gmail</a>
            </li>

            <li>
            <i class="fa fa-calendar" aria-hidden="true"></i> TuTh
            </li>

            <li>
            <i class="fa fa-clock-o" aria-hidden="true"></i> 13:15-14:30
            </li>

            <li>
                <a href=""><i class="fa fa-university" aria-hidden="true"></i> Zoom</a>
            </li>
            

            <li>
            <a href="https://twitter.com/hallettmichael"><i class="fa fa-twitter" aria-hidden="true"></i></a> 
            </li>

            <li>
            <a href="https://github.com/https://github.com/hallettmiket/hallettmiket.github.io"><i class="fa fa-github" aria-hidden="true"></i></a>
            </li>
            <span class="fake-li"><i class="fa fa-slack" aria-hidden="true"></i> <a href="https://biol480-18-concordia.slack.com/">Slack</a></span><br>

            
        </ul>

        <div class="credits">
        <span>All content licensed under a <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons</a>.
        Content © 2020 <a href="https://mikehallett.science">mike hallett</a>
        </span><br><br>
<br><br>
        <span>We thank <a href="https://www.andrewheiss.com/">A Heiss</a> for the modified Tufte scheme.</span><br><br>

        <span>This site modified the <a href="https://github.com/andrewheiss/ath-tufte-hugo">ath-tufte-hugo theme</a> in <a href="https://bookdown.org/yihui/blogdown/">blogdown</a> and <a href="https://gohugo.io/">hugo</a>.</span>
        </div>
    </footer>

    
    <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
      }
    });
    </script>
    <script async type="text/javascript"
      src="https://cdn.bootcss.com/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>


</div>
</body>
</html>

    <link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css"/>
